libpod/container_internal_linux.go

```diff
@@ -205,6 +205,50 @@ func (c *Container) resolveWorkDir() error {
 		// the path exists on the container.
 		if err != nil {
 			if os.IsNotExist(err) {
+				// We cannot create workdir since explicit --workdir is
+				// set in config but workdir could also be a symlink.
+				// If its a symlink lets check if resolved link is present
+				// on the container or not.
+
+				// If we can resolve symlink and resolved link is present on the container
+				// then return nil cause its a valid use-case.
+
+				maxSymLinks := 0
+				for {
+					// Linux only supports a chain of 40 links.
+					// Reference: https://github.com/torvalds/linux/blob/master/include/linux/namei.h#L13
+					if maxSymLinks > 40 {
+						break
+					}
+					resolvedSymlink, err := os.Readlink(resolvedWorkdir)
+					if err != nil {
+						// End sym-link resolution loop.
+						break
+					}
+					if err == nil && resolvedSymlink != "" {
+						_, resolvedSymlinkWorkdir, err := c.resolvePath(c.state.Mountpoint, resolvedSymlink)
+						if isPathOnVolume(c, resolvedSymlinkWorkdir) || isPathOnBindMount(c, resolvedSymlinkWorkdir) {
+							// Resolved symlink exists on external volume or mount
+							return nil
+						}
+						if err != nil {
+							// Could not resolve path so end sym-link resolution loop.
+							break
+						}
+						if err == nil && resolvedSymlinkWorkdir != "" {
```

Comment:
no need to check for `err == nil` here, we already have a check for `!= nil` that breaks the loop